site/src/pages/TemplatesPage/TemplatesPageView.tsx

```diff
@@ -114,94 +124,108 @@ export const TemplatesPageView: FC<React.PropsWithChildren<TemplatesPageViewProp
-              <TableCell width="16%">{Language.createdByLabel}</TableCell>
-              <TableCell width="1%"></TableCell>
-            </TableRow>
-          </TableHead>
-          <TableBody>
-            {props.loading && <TableLoader />}
-            {!props.loading && !props.templates?.length && (
+      {props.getOrganizationsError ? (
+        <ErrorSummary
+          error={props.getOrganizationsError}
+          defaultMessage={t("errors.getOrganizationsError")}
+        />
+      ) : props.getTemplatesError ? (
+        <ErrorSummary
+          error={props.getTemplatesError}
+          defaultMessage={t("errors.getTemplatesError")}
+        />
+      ) : (
+        <TableContainer>
+          <Table>
+            <TableHead>
               <TableRow>
-                <TableCell colSpan={999}>
-                  <EmptyState
-                    message={Language.emptyMessage}
-                    description={
-                      props.canCreateTemplate
-                        ? Language.emptyDescription
-                        : Language.emptyViewNoPerms
-                    }
-                    descriptionClassName={styles.emptyDescription}
-                    cta={<CodeExample code="coder templates init" />}
-                  />
-                </TableCell>
+                <TableCell width="50%">{Language.nameLabel}</TableCell>
+                <TableCell width="16%">{Language.usedByLabel}</TableCell>
+                <TableCell width="16%">{Language.lastUpdatedLabel}</TableCell>
+                <TableCell width="16%">{Language.createdByLabel}</TableCell>
+                <TableCell width="1%"></TableCell>
               </TableRow>
-            )}
-            {props.templates?.map((template) => {
-              const templatePageLink = `/templates/${template.name}`
-              const hasIcon = template.icon && template.icon !== ""
-
-              return (
-                <TableRow
-                  key={template.id}
-                  hover
-                  data-testid={`template-${template.id}`}
-                  tabIndex={0}
-                  onKeyDown={(event) => {
-                    if (event.key === "Enter") {
-                      navigate(templatePageLink)
-                    }
-                  }}
-                  className={styles.clickableTableRow}
-                >
-                  <TableCellLink to={templatePageLink}>
-                    <AvatarData
-                      title={template.name}
-                      subtitle={template.description}
-                      highlightTitle
-                      avatar={
-                        hasIcon ? (
-                          <div className={styles.templateIconWrapper}>
-                            <img alt="" src={template.icon} />
-                          </div>
-                        ) : undefined
+            </TableHead>
+            <TableBody>
+              {props.loading && <TableLoader />}
+              {empty && (
```

Comment:
What do you think about making a parent table component that accepts a child table body component? That child component could be a loading component, or a data-filled component. I was wondering if breaking apart state into components here might make this a little easier to read and reason about.

Reply:
Sounds like a good idea, can you make a ticket for it? We could apply it in multiple places.